File: java/cours/src/session3/TD/MotRecursif/MotVideInductif.java
package session3.td;

public class MotVideInductif implements Mot {
	public static MotVideInductif VIDE = new MotVideInductif();
	
	private MotVideInductif() {}

	@Override
	public String getValue() {
		return "";
	}

	@Override
	public int taille() {
		return 0;
	}
	
	@Override
	public boolean estVide() {
		return true;
	}
}

File: typescript/session3/MotRecursif/MotVideInductif.ts
import { Mot } from "./mot";

export class MotVideInductif extends Mot {

  public static VIDE: MotVideInductif;

  private constructor() {
    super();
  }

  public static getInstance(): MotVideInductif {
    if (this.VIDE == null) {
      this.VIDE = new MotVideInductif;
    }
    return this.VIDE;
  }

  public getValue(): string {
    return '';
  }

  public taille(): number {
    return 0;
  }

  public estVide(): boolean {
    return true;
  }

}